### PR TITLE
Datatable: Hideable columns

### PR DESCRIPTION
**Pull request by ClockworkOnion**
When displaying the datatable, a dropdown menu allows for hiding/revealing individual columns
image
Selections persist when moving/zooming the map, but not when switching to a different layer/datatable, since the table component is recreated in this case.

(Also, coordinate preview in the table is cut off after 5 places (including dot), full display available via button)
image

### CODE REVIEW

**Comment by ClockworkOnion** [comment]
[Ausblenden von Spalten in der Tabelle](https://trello.com/c/0QBA7e3t/1019-ausblenden-von-spalten-in-der-tabelle)

File: projects/core/src/lib/datatable/table/table.component.html [hunk 1]
```diff
@@ -66,5 +66,12 @@
...
-<mat-paginator [pageSizeOptions]="[5, 10, 20]" showFirstLastButtons></mat-paginator>
+<mat-form-field id="columnSelector" class="dropdown">
+    <mat-label>Select columns</mat-label>
+    <mat-select multiple [(ngModel)]="displayedColumns">
+        <mat-option *ngFor="let columnName of checkboxLabels" [value]="columnName" (click)="showCheckboxColumn()">
+            {{ fixedColumnDescriptors(columnName) }}
+        </mat-option>
+    </mat-select>
+</mat-form-field>
+<mat-paginator class="paginate" [pageSizeOptions]="[5, 10, 20]" showFirstLastButtons></mat-paginator>
```
Comment (jdroenner) [inline projects/core/src/lib/datatable/table/table.component.html:77]: why do you need the paginate class?

File: projects/core/src/lib/datatable/table/table.component.ts [hunk 2]
```diff
@@ -131,11 +132,37 @@ export class DataTableComponent implements OnInit, AfterViewInit, OnDestroy, OnC
         this.dataSource.data = data.data;
         this.featureColumns = metadata.dataTypes.keySeq().toArray();
         this.featureColumnDataTypes = metadata.dataTypes.valueSeq().toArray();
-        this.displayedColumns = ['_____select', '_____coordinates', '_____table__start', '_____table__end'].concat(this.featureColumns);
+        if (this.displayedColumns.length === 0) {
```
Comment (jdroenner) [inline projects/core/src/lib/datatable/table/table.component.ts:135]: what case is this?

File: projects/core/src/lib/datatable/table/table.component.ts [hunk 3]
```diff
@@ -131,11 +132,37 @@ export class DataTableComponent implements OnInit, AfterViewInit, OnDestroy, OnC
...
+            this.displayedColumns = ['_____select', '_____coordinates', '_____table__start', '_____table__end'].concat(this.featureColumns);
+        }
         this.featureColumnDataTypes = this.getColumnProperties();
+        this.checkboxLabels = ['_____coordinates', '_____table__start', '_____table__end'].concat(this.featureColumns);
         setTimeout(() => this.navigatePage(this.projectService.getSelectedFeature()));
     }
 
+    showCheckboxColumn(): void {
```
Comment (jdroenner) [inline projects/core/src/lib/datatable/table/table.component.ts:143]: please add some info what this does and maybe use a more descriptive name

File: projects/core/src/lib/datatable/table/table.component.html (hunk 1 above)
Comment (ClockworkOnion) [reply projects/core/src/lib/datatable/table/table.component.html]: Right, I adressed the mat-paginator directly inside the css, that's cleaner 75328626bcc97511d5f50b686668d3dbb8dfaa6c

File: projects/core/src/lib/datatable/table/table.component.ts (hunk 3 above)
Comment (ClockworkOnion) [reply projects/core/src/lib/datatable/table/table.component.ts]: I changed the name and added some info above the method

**Review by jdroenner — APPROVED** [review]

**Comment by jdroenner** [comment]
bors merge

**Pull request merged**